Restore CI warnings-as-errors for FirebaseAuth pod tests with Xcode 16

### Description

    - ERROR | xcodebuild:  /Users/runner/work/firebase-ios-sdk/firebase-ios-sdk/FirebaseAuth/Sources/Swift/Utilities/AuthWebViewController.swift:19:3: error: add '@preconcurrency' to suppress 'Sendable'-related warnings from module 'WebKit'

### Reproducing the issue

See workaround in FirebaseAuth.podspec and https://github.com/firebase/firebase-ios-sdk/actions/runs/11003726045/job/30553328769#step:7:110

### Firebase SDK Version

11.2

### Xcode Version

16.0

### Installation Method

CocoaPods

### Firebase Product(s)

Authentication

### Targeted Platforms

iOS

### Relevant Log Output

_No response_

### If using Swift Package Manager, the project's Package.resolved

<!--- Look below for instructions on how to share your Package.resolved. --->

<details>
<summary>Expand <code>Package.resolved</code> snippet</summary>
<br>

```json

Replace this line with the contents of your Package.resolved.

```

</details>


### If using CocoaPods, the project's Podfile.lock

<!--- Look below for instructions on how to share your Podfile.lock. --->

<details>
<summary>Expand <code>Podfile.lock</code> snippet</summary>
<br>

```yml

Replace this line with the contents of your Podfile.lock!

```

</details>

--- a/FirebaseAuth.podspec
+++ b/FirebaseAuth.podspec
@@ -48,13 +48,11 @@ supports email and password accounts, as well as several 3rd party authenticatio
     'FirebaseAuth/README.md',
     'FirebaseAuth/CHANGELOG.md'
   ]
-  # TODO(#13704) Restore warnings-as-errors checking.
-  #    'OTHER_SWIFT_FLAGS' => "$(inherited) #{ENV.key?('FIREBASE_CI') ? '-D FIREBASE_CI -warnings-as-errors' : ''}"
   s.pod_target_xcconfig = {
     'GCC_C_LANGUAGE_STANDARD' => 'c99',
     # The second path is to find FirebaseAuth-Swift.h from a pod gen project
     'HEADER_SEARCH_PATHS' => '"${PODS_TARGET_SRCROOT}" "${OBJECT_FILE_DIR_normal}/${NATIVE_ARCH_ACTUAL}"',
-    'OTHER_SWIFT_FLAGS' => "$(inherited) #{ENV.key?('FIREBASE_CI') ? '-D FIREBASE_CI' : ''}"
+    'OTHER_SWIFT_FLAGS' => "$(inherited) #{ENV.key?('FIREBASE_CI') ? '-D FIREBASE_CI -warnings-as-errors' : ''}"
   }
   s.framework = 'Security'
   s.ios.framework = 'SafariServices'

--- a/FirebaseAuth/Tests/Unit/SwiftAPI.swift
+++ b/FirebaseAuth/Tests/Unit/SwiftAPI.swift
@@ -365,7 +365,7 @@ class AuthAPI_hOnlyTests: XCTestCase {
       @available(iOS 13, tvOS 13, macOS 10.15, macCatalyst 13, watchOS 7, *)
       func FIRFedederatedAuthProvider_hAsync() async throws {
         let obj = FederatedAuthImplementation()
-        try await _ = obj.credential(with: nil)
+        _ = try await obj.credential(with: nil)
       }
     }
   #endif

